### PR TITLE
(#19) Mock subject version endpoints

### LINKED ISSUE #19: Schema Registry List and Get subject versions
I would like to be able register multiple schemas under a subject, and know about their evolution. 

Add ability to perform

```
GET /subjects/:name/versions/
GET /subjects/:name/versions/:versionId
GET /subjects/:name/versions/latest
```

### PR DESCRIPTION
**Pull request by cricket007**
Closes #19

### CODE REVIEW

File: schema-registry-mock/src/main/java/com/bakdata/schemaregistrymock/SchemaRegistryMock.java [hunk 1]
```diff
@@ -84,9 +87,12 @@
     private static final String SCHEMA_REGISTRATION_PATTERN = "/subjects/[^/]+/versions";
```
Comment (AHeise) [inline schema-registry-mock/src/main/java/com/bakdata/schemaregistrymock/SchemaRegistryMock.java]: Please rename the constant to something like SCHEMA_PATH_PATTERN.

**Comment by cricket007** [comment]
> add a test to retrieve the 'latest' version of a schema?

That's what `getLatestSchemaMetadata` is testing 

But if I understand correctly, you want to make sure it doesn't just always work for a single schema

**Comment by AHeise** [comment]
Yes, sorry it was poorly phrased. It would be good to actually add two versions and ensure that we get the latest.

**Comment by cricket007** [comment]
@AHeise Good to go?

**Comment by AHeise** [comment]
Thank you very much. I currently need to address another issue and will do a release afterwards.

**Pull request merged**